### ISSUE TITLE
Incompatible checks skipped and want to disable them

### ISSUE THREAD

**Issue by glenwinters** [rrrene/credo#680]
### Environment

* Credo version (`mix credo -v`): 1.1.1
* Erlang/Elixir version (`elixir -v`): 1.8.2
* Operating system: macOS 10.14.5

### What were you trying to do?

I'm running credo to look for issues. I am seeing an `info` level warning about skipping incompatible checks. However, I don't have those checks listed in my config, so I'm not sure why they're running.

### Expected outcome

No warnings about skipped messages.

### Actual outcome

```
info: the following checks were skipped because they're not compatible with
your version of Elixir (1.8.2).

   1) Credo.Check.Refactor.MapInto
   2) Credo.Check.Warning.LazyLogging


Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 3.9 seconds (0.07s to load, 3.9s running checks)
885 mods/funs, found no issues.

Use `--strict` to show all issues, `--help` for options.
```

### Credo config

```
%{
  configs: [
    %{
      name: "default",
      color: true,
      strict: false,
      check_for_updates: true,
      files: %{
        included: ["lib/", "config/"],
        excluded: []
      },
      checks: [
        {Credo.Check.Consistency.ExceptionNames},
        {Credo.Check.Consistency.LineEndings},
        {Credo.Check.Consistency.ParameterPatternMatching},
        {Credo.Check.Consistency.SpaceAroundOperators},
        {Credo.Check.Consistency.SpaceInParentheses},
        {Credo.Check.Consistency.TabsOrSpaces, priority: :higher},
        {Credo.Check.Design.TagTODO},
        {Credo.Check.Design.TagFIXME},
        {Credo.Check.Design.DuplicatedCode, excluded_macros: []},
        {Credo.Check.Design.AliasUsage, priority: :low},
        {Credo.Check.Readability.AliasOrder},
        {Credo.Check.Readability.FunctionNames},
        {Credo.Check.Readability.LargeNumbers},
        {Credo.Check.Readability.MaxLineLength, priority: :normal, max_length: 100},
        {Credo.Check.Readability.ModuleAttributeNames},
        {Credo.Check.Readability.ModuleDoc},
        {Credo.Check.Readability.ModuleNames},
        {Credo.Check.Readability.ParenthesesOnZeroArityDefs},
        {Credo.Check.Readability.ParenthesesInCondition},
        {Credo.Check.Readability.PredicateFunctionNames},
        {Credo.Check.Readability.PreferImplicitTry},
        {Credo.Check.Readability.RedundantBlankLines},
        {Credo.Check.Readability.StringSigils},
        {Credo.Check.Readability.TrailingBlankLine},
        {Credo.Check.Readability.TrailingWhiteSpace},
        {Credo.Check.Readability.VariableNames},
        {Credo.Check.Readability.Semicolons},
        {Credo.Check.Readability.SpaceAfterCommas},
        {Credo.Check.Refactor.DoubleBooleanNegation},
        {Credo.Check.Refactor.CondStatements},
        {Credo.Check.Refactor.CyclomaticComplexity},
        {Credo.Check.Refactor.FunctionArity},
        {Credo.Check.Refactor.LongQuoteBlocks},
        {Credo.Check.Refactor.MatchInCondition},
        {Credo.Check.Refactor.NegatedConditionsInUnless},
        {Credo.Check.Refactor.NegatedConditionsWithElse},
        {Credo.Check.Refactor.Nesting},
        {Credo.Check.Refactor.PipeChainStart},
        {Credo.Check.Refactor.UnlessWithElse},
        {Credo.Check.Refactor.WithClauses, priority: :low},
        {Credo.Check.Warning.BoolOperationOnSameValues},
        {Credo.Check.Warning.IExPry},
        {Credo.Check.Warning.IoInspect},
        {Credo.Check.Warning.OperationOnSameValues},
        {Credo.Check.Warning.OperationWithConstantResult},
        {Credo.Check.Warning.UnusedEnumOperation},
        {Credo.Check.Warning.UnusedFileOperation},
        {Credo.Check.Warning.UnusedKeywordOperation},
        {Credo.Check.Warning.UnusedListOperation},
        {Credo.Check.Warning.UnusedPathOperation},
        {Credo.Check.Warning.UnusedRegexOperation},
        {Credo.Check.Warning.UnusedStringOperation},
        {Credo.Check.Warning.UnusedTupleOperation},
        {Credo.Check.Warning.RaiseInsideRescue}
      ]
    }
  ]
}
```

**Comment by rrrene** [comment]
> I don't have those checks listed in my config, so I'm not sure why they're running.

Credo merges your config with its default config to avoid any missing config-keys causing any problems. We merge the checks as well as their parameters, so that we can make changes to default parameters (unless you have specified your own value for a parameter). This is also the way we introduce new checks in minor version bumps.

Basically, disabling the checks will help you get rid of the `info` message.

I just had a look and you are right in that it is a "warning" since it goes to `stderr`. This seems wrong. I will fix this and rephrase the `info` message to convey how to get rid of it. 

**TL;DR** Thanks for reporting this! :+1:

**Comment by TheFirstAvenger** [comment]
I believe this can be closed as it was addressed in https://github.com/rrrene/credo/commit/cb293291d518ea38c6e732dc04504b0bb2e8e06a

**Comment by glenwinters** [comment]
@TheFirstAvenger Thanks. Looks like that fix was released in 1.1.3.
